src/arr/base/contracts.arr

```diff
@@ -33,13 +40,17 @@ data FieldFailure:
   | missing-field(loc, field) with:
     render-reason(self, loc, from-fail-arg):
       [ED.error:
-        [ED.para: ED.text("Missing field"), ED.code(ED.text(self.field)),
-          ED.text("is required at"), draw-and-highlight(self.loc)]]
+        [ED.para: ED.text("Missing field "), ED.code(ED.text(self.field)),
```

Comment:
`ED.text` within an `ED.para` shouldn't need trailing whitespace.  Either use `ED.para-nospace` or remove the whitespace, here and everywhere.